RapidReact/src/main/java/frc/robot/subsystem/ClimberSubsystem.java

```diff
@@ -1,114 +1,503 @@
+    // climberRight.configClosedLoopPeriod(1, closedLoopTimeMs);
+
+    /**
+     * configAuxPIDPolarity(boolean invert, int timeoutMs)
+     * false means talon's local output is PID0 + PID1, and other side Talon is PID0 - PID1
+     * true means talon's local output is PID0 - PID1, and other side Talon is PID0 + PID1
+     */
+    // climberRight.configAuxPIDPolarity(false);
+
+    /* Initialize */
+    // climberRight.setStatusFramePeriod(StatusFrameEnhanced.Status_10_Targets, 10);
+
+    // zero sensors
+    climberLeft.getSensorCollection().setQuadraturePosition(0, MotorUtils.CONTROLLER_TIMEOUT_MS);
+    climberRight.getSensorCollection().setQuadraturePosition(0, MotorUtils.CONTROLLER_TIMEOUT_MS);
+
+    if (Robot.isSimulation()) {
+      CTREPhysicsSim.getInstance().addTalonSRX(climberLeft, .75, 5100, false);
+      CTREPhysicsSim.getInstance().addTalonSRX(climberRight, .75, 5100, false);
+    }
     if (config.enablePneumatics) {
-      if (Robot.isSimulation()) {
+      if (Robot.isSimulation())
+      {
         elevatorSolenoid = new DoubleSolenoid(PneumaticsModuleType.CTREPCM, config.elevatorSolenoid_ID1, config.elevatorSolenoid_ID2);
       }
-      else {
+      else
+      {
         elevatorSolenoid = new DoubleSolenoid(PneumaticsModuleType.REVPH, config.elevatorSolenoid_ID1, config.elevatorSolenoid_ID2);
       }
     }
+
+    currentClimbState = ClimbState.Idle;
+  }
+
+  boolean isClimberAtSetpoint(double setpoint) {
+    boolean climberLeftAtSetpoint = false;
+
+    // if it's within the error threshold for a set amount of loops
+    if (
+      climberLeft.getClosedLoopError() < +climbErrThreshold && climberLeft.getClosedLoopError() > -climbErrThreshold
+    ) {
+      withinThresholdLoops1++;
+      if (withinThresholdLoops1 > climbLoopsToSettle) {
+        // and ifs within the error threshold of the position
+        if (
+          climberLeft.getSelectedSensorPosition() > setpoint - climbErrThreshold &&
+          climberLeft.getSelectedSensorPosition() < setpoint + climbErrThreshold
+        ) {
+          // then it's reached
+          climberLeftAtSetpoint = true;
+        }
+      }
+    } else {
+      withinThresholdLoops1 = 0;
+    }
+
+    boolean climberRightAtSetpoint = false;
+
+    if (
+      climberRight.getClosedLoopError() < +climbErrThreshold &&
+      climberRight.getClosedLoopError() > -climbErrThreshold
+    ) {
+      withinThresholdLoops2++;
+      if (withinThresholdLoops2 > climbLoopsToSettle) {
+        if (
+          climberRight.getSelectedSensorPosition() > setpoint - climbErrThreshold &&
+          climberRight.getSelectedSensorPosition() < setpoint + climbErrThreshold
+        ) {
+          climberRightAtSetpoint = true;
+        }
+      }
+    } else {
+      withinThresholdLoops2 = 0;
+    }
+
+    return climberLeftAtSetpoint && climberRightAtSetpoint;
+  }
+
+  private void idleInit() {
+    // setElevatorTilted(false);
```

Comment:
Why have you taken the pneumatics out of these? If you need to disable them in testing, it would be more appropriate to do so in the setElevatorTilted function.